From e3665748e6ff6b9514a54f6341583afa88511682 Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Goudagunta <gpradeep@codeaurora.org>
Date: Fri, 25 Jul 2014 11:41:19 +0530
Subject: [PATCH] wcnss: add null check in pm_ops unregister

Add null check for pm_ops in pm_ops unregister to avoid
null pointer dereference.

Change-Id: Ifdad0a0fecc9b8cd97910e4f491ee08ae4a58483
CRs-Fixed: 690191
Signed-off-by: Pradeep Kumar Goudagunta <gpradeep@codeaurora.org>
---
 drivers/net/wireless/wcnss/wcnss_wlan.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/wireless/wcnss/wcnss_wlan.c b/drivers/net/wireless/wcnss/wcnss_wlan.c
index 15e79d5..db09d02 100644
--- a/drivers/net/wireless/wcnss/wcnss_wlan.c
+++ b/drivers/net/wireless/wcnss/wcnss_wlan.c
@@ -1409,6 +1409,12 @@ void wcnss_wlan_unregister_pm_ops(struct device *dev,
 				const struct dev_pm_ops *pm_ops)
 {
 	if (penv && dev && (dev == &penv->pdev->dev) && pm_ops) {
+		if (penv->pm_ops == NULL) {
+			pr_err("%s: pm_ops is already unregistered.\n",
+				 __func__);
+			return;
+		}
+
 		if (pm_ops->suspend != penv->pm_ops->suspend ||
 				pm_ops->resume != penv->pm_ops->resume)
 			pr_err("PM APIs dont match with registered APIs\n");
-- 
1.8.2.1

